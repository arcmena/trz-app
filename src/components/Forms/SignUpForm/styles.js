import styled from 'styled-components';

export const Form = styled.form`
    display: flex;
    flex-direction: column;
    align-items: center;

    label {
        font-family: 'Ubuntu';
        font-weight: lighter;
        font-size: 19px;
        margin-bottom: 5px;
        float: left;
    }

    div {
        margin-bottom: 10px;
    }

    div:first-child {
        flex-direction: column;
    }

    @media (max-width: 430px) {
        input {
            width: 150px;
        }
    }
`;

export const Name = styled.div`
    display: flex;
    flex-direction: column;

    @media (max-width: 430px) {
        div {
            width: 320px;
            input {
                width: 310px;
            }
        }
    }
`;

export const Title = styled.h2`
    font-family: 'PT Sans';
    font-size: 35px;

    color: #383838;

    margin-bottom: 15px;

    span {
        color: #61c1d4;
    }
`;

export const DivRow = styled.div`
    display: flex;
    justify-content: space-between;

    width: 430px;

    div {
        width: 200px;
        margin-bottom: 10px;

        input {
            width: 190px;
        }
    }

    select {
        display: flex;
        align-items: center;

        width: 200px;
        height: 60px;
        border: 2px solid rgba(48, 48, 48, 0.35);
        border-radius: 15px;

        padding: 8px 10px;

        font-family: 'Ubuntu';
        font-size: 20px;
    }

    @media (max-width: 430px) {
        flex-direction: column;

        width: 320px;

        div {
            width: 320px;
            input {
                width: 310px;
            }
        }

        select {
            width: 320px;
        }
    }
`;

export const ItemsDiv = styled.div`
    display: flex;
    flex-direction: column;

    margin: 0 !important;

    span {
        font-family: 'Ubuntu';
        font-weight: lighter;
        font-size: 19px;
        margin-bottom: 5px;
    }

    @media (max-width: 500px) {
        width: 350px;
        overflow-x: scroll;
        flex-shrink: 0;
        flex-wrap: wrap;
    }
`;
