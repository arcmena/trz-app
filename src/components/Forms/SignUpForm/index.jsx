import React from 'react';

import { Input, Button, Items } from '../../';
import { Form, Name, Title, DivRow, ItemsDiv } from './styles';

export default ({ onChange, onSubmit }) => {
    return (
        <Form onChange={onChange} onSubmit={onSubmit}>
            <Title>
                Sign Up <span>Survivor!</span>
            </Title>

            <Name>
                <label htmlFor="name">Name</label>
                <Input type="text" id="name" name="name" />
            </Name>

            <DivRow>
                <div>
                    <label htmlFor="age">Birth Date</label>
                    <Input type="date" id="age" name="age" />
                </div>
                <div>
                    <label htmlFor="gender">Gender</label>
                    <select id="gender" name="gender">
                        <option value="" defaultChecked>
                            Select...
                        </option>
                        <option value="M">Male</option>
                        <option value="F">Female</option>
                    </select>
                </div>
            </DivRow>

            <ItemsDiv>
                <span>Register belongings</span>
                <Items />
            </ItemsDiv>

            <Button oulined>
                <span>FINISH</span>
            </Button>
        </Form>
    );
};
