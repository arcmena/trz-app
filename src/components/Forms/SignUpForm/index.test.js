import React from 'react';
import { shallow } from 'enzyme';
import SignUpForm from './';
import { Button } from '../../';

const essentialProps = {
    onChange: jest.fn(),
    onClick: jest.fn(),
    onChangeItems: jest.fn(),
};

describe('<SignUpForm/>', () => {
    it('should render as expected', () => {
        const wrapper = shallow(<SignUpForm {...essentialProps} />);
        expect(wrapper).toMatchSnapshot();
    });
    it('should render the form', () => {
        const wrapper = shallow(<SignUpForm {...essentialProps} />);
        essentialProps.onChange.mockReset();
        expect(wrapper.find('form').props()).toEqual({
            onChange: essentialProps.onChange,
        });

        expect(wrapper.find('h2').text()).toBe('Sign Up');
        expect(wrapper.find('span').text()).toBe('Survivor!');
        expect(wrapper.find('label').props()).toEqual({
            htmlFor: 'name',
        });
    });
});
