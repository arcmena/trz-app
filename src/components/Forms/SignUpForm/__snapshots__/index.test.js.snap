// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm/> should render as expected 1`] = `
<div>
  <styled.form
    onChange={[MockFunction]}
  >
    <styled.h2>
      Sign Up 
      <span>
        Survivor!
      </span>
    </styled.h2>
    <styled.div>
      <label
        htmlFor="name"
      >
        Name
      </label>
      <_default
        id="name"
        name="name"
        required={true}
        type="text"
      />
    </styled.div>
    <styled.div>
      <div>
        <label
          htmlFor="age"
        >
          Age
        </label>
        <_default
          id="age"
          name="age"
          required={true}
          type="text"
        />
      </div>
      <div>
        <label
          htmlFor="gender"
        >
          Gender
        </label>
        <select
          id="gender"
          name="gender"
          required={true}
        >
          <option
            defaultChecked={true}
            value=""
          >
            Select...
          </option>
          <option
            value="M"
          >
            Male
          </option>
          <option
            value="F"
          >
            Female
          </option>
        </select>
      </div>
    </styled.div>
  </styled.form>
  <styled.div
    onChange={[MockFunction]}
  >
    <span>
      Register belongings
    </span>
    <_default
      borderColor="rgba(48, 48, 48, 0.35)"
      borderTop={true}
    />
  </styled.div>
  <styled.div>
    <styled.button
      onClick={[MockFunction]}
      oulined={true}
      type="submit"
    >
      <span>
        FINISH
      </span>
    </styled.button>
  </styled.div>
</div>
`;

exports[`<SignUpForm/> should render as expected 1`] = `
<div>
  <styled.form
    onChange={[MockFunction]}
  >
    <styled.h2>
      Sign Up 
      <span>
        Survivor!
      </span>
    </styled.h2>
    <styled.div>
      <label
        htmlFor="name"
      >
        Name
      </label>
      <_default
        id="name"
        name="name"
        required={true}
        type="text"
      />
    </styled.div>
    <styled.div>
      <div>
        <label
          htmlFor="age"
        >
          Age
        </label>
        <_default
          id="age"
          name="age"
          required={true}
          type="text"
        />
      </div>
      <div>
        <label
          htmlFor="gender"
        >
          Gender
        </label>
        <select
          id="gender"
          name="gender"
          required={true}
        >
          <option
            defaultChecked={true}
            value=""
          >
            Select...
          </option>
          <option
            value="M"
          >
            Male
          </option>
          <option
            value="F"
          >
            Female
          </option>
        </select>
      </div>
    </styled.div>
  </styled.form>
  <styled.div
    onChange={[MockFunction]}
  >
    <span>
      Register belongings
    </span>
    <_default
      borderColor="rgba(48, 48, 48, 0.35)"
      borderTop={true}
    />
  </styled.div>
  <styled.div>
    <styled.button
      onClick={[MockFunction]}
      oulined={true}
      type="submit"
    >
      <span>
        FINISH
      </span>
    </styled.button>
  </styled.div>
</div>
`;
